src/main/java/mate/academy/spring/config/SecurityConfig.java

```diff
@@ -0,0 +1,47 @@
+package mate.academy.spring.config;
+
+import java.security.SecureRandom;
+import java.util.Properties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.provisioning.InMemoryUserDetailsManager;
+
+@EnableWebSecurity
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
+    @Override
+    protected void configure(AuthenticationManagerBuilder auth) throws Exception {
+        auth.userDetailsService(inMemoryUserDetailsManager());
+    }
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http
+                .authorizeRequests()
+                .anyRequest().authenticated()
+                .and()
+                .formLogin()
+                .permitAll()
+                .and()
+                .httpBasic()
+                .and()
+                .csrf().disable();
+    }
+
+    @Bean
+    public InMemoryUserDetailsManager inMemoryUserDetailsManager() {
+        Properties users = new Properties();
+        users.put("bob@domain.com", getEncoder().encode("1234") + ",ROLE_USER,enabled");
+        users.put("alice@domain.com", getEncoder().encode("1234") + ",ROLE_USER,enabled");
+        return new InMemoryUserDetailsManager(users);
+    }
+
+    @Bean
+    public PasswordEncoder getEncoder() {
+        return new BCryptPasswordEncoder(10, new SecureRandom());
+    }
```

Comment:
if you move this method to another config class
add field here with this bean
you will use just 1 object, instead of a new object for each method invocation

Reply:
Thanks! :thumbsup: